frontend/src/services/download.js

```diff
@@ -0,0 +1,60 @@
+function androidSaveBase64(filename, blob, mimetype) {
+  // SomenergiaPlatform object is injected in mobile applications.
+  // It offers a hook to regular downloads which are diabled by default.
+  if (!window.SomenergiaPlatform) return
+  // We need to convert the blob to base64
+  var reader = new FileReader()
+  reader.readAsDataURL(blob)
+  reader.onloadend = function () {
+    // Remove trailing url header
+    let base64String = reader.result.replace(/[^,]*,/, '')
+    window.SomenergiaPlatform.saveBase64(filename, base64String, mimetype)
+  }
+  return true
+}
+
+function androidSaveUtf8(filename, text, mimetype) {
+  // SomenergiaPlatform object is injected in mobile applications.
+  // It offers a hook to regular downloads which are diabled by default.
+  if (!window.SomenergiaPlatform) return false
+  window.SomenergiaPlatform.saveUtf8(filename, text, mimetype)
+  return true
+}
+
+/// Forces a download by creating a temporary download link and clicking it
+function downloadUrl(filename, url, mimetype) {
+  console.log({downloadUrl, filename, url, mimetype})
+  const link = document.createElement('a')
+  link.href = url
+  if (filename) link.download = filename
+  document.body.appendChild(link)
+  link.click()
+  // clean up
+  document.body.removeChild(link)
+}
+
+/// Platform sensible download of a binary file
+function downloadBlob(filename, blob, mimetype='application/octet-stream') {
+  if (androidSaveBase64(filename, blob, mimetype)) return blob
+  const url = window.URL.createObjectURL(new Blob([blob]))
+  downloadUrl(filename, url, mimetype)
+  URL.revokeObjectURL(url)
+  return blob
+}
+
+/// Platform sensible download of a text based file
+function downloadTextFile(filename, text, mimetype='text/plain') {
+  if (androidSaveUtf8(filename, text, mimetype)) return text
+  const url = `data:${mimetype};charset=utf-8,` + encodeURIComponent(text)
+  downloadUrl(filename, url, mimetype)
+}
+
+/// Platform sensible download of a zip based file
+function downloadZipFile(filename, blob, mimetype='application/zip') {
```

Comment:
downloadBlobFile was abstracted to work also with Zip files just changing the mimetype parameter. Did you have any problems using it?

At the end we have a binary and a text downloader. The strategies are different because binary has to use a blob, while text can be passed as is.